src/main/java/parser/ParserFactory.java

```diff
@@ -0,0 +1,206 @@
+package parser;
+
+import command.AddCommand;
+import command.Command;
+import command.CommentCommand;
+import command.DeleteCommand;
+import command.DoneCommand;
+import command.ExitCommand;
+import command.FindCommand;
+import command.ListCommand;
+import command.LocationCommand;
+import command.PostponeCommand;
+import command.PriorityCommand;
+import command.RemindCommand;
+import command.SearchCommand;
+import command.ViewCommand;
+import command.EditCommand;
+import command.IgnoreCommand;
+import exception.DukeException;
+import task.Deadline;
+import ui.Ui;
+
+import java.time.LocalDateTime;
+import java.text.ParseException;
+import java.util.Date;
+
+/**
+ * The parser class is used to parse and make sense of the different queries the
+ * user inputs into the program and tag them for further processing.
+ *
+ * @author Sai Ganesh Suresh
+ * @version v2.0
+ */
+
+public class ParserFactory {
+
+    /**
+     * Parses the user input of string type and returns the respective command type.
+     *
+     * @param userInput This string is provided by the user to ask 'Duke' to perform
+     *                  a particular action
+     * @return Command After processing the user's input it returns the correct
+     *         command for further processing
+     * @throws DukeException The DukeException class has all the respective methods
+     *                       and messages!
+     */
+    public static Command parse(String userInput) throws DukeException {
```

Comment:
Too lengthy to read and follow what it does. Try to simplify this by extracting out low-level operation to methods.